### PR TITLE
feat(dashmate): report pulled images during update

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

Dashmate should show info about the images of the service that has been pull during update https://github.com/dashpay/platform/issues/105

## What was done?

* `dashmate update` command now states which images were updating during pull
* Added `--format` flag to the `dashmate` update command
* Fixed incorrect node services running check

## How Has This Been Tested?

* Locally
* CI
* Devnet

Reset and status command also affected by this PR and were tested as well.

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by pshenmic** [comment]
Test results:

Default output:

image

Format json:

```
$ dashmate update --format=json
[{"serviceName":"core","title":"Core","pulled":false,"image":"dashpay/dashd:19.2.0"},{"serviceName":"drive_abci","title":"Drive ABCI","pulled":false,"image":"dashpay/drive:0.24"},{"serviceName":"drive_tenderdash","title":"Drive Tenderdash","pulled":true,"image":"dashpay/tenderdash:0.11.2"},{"serviceName":"dapi_api","title":"DAPI API","pulled":false,"image":"dashpay/dapi:0.24"},{"serviceName":"dapi_envoy","title":"DAPI Envoy","pulled":false,"image":"dashpay/envoy:0.24"},{"serviceName":"dashmate_helper","title":"Dashmate Helper","pulled":false,"image":"dashpay/dashmate-helper:0.24"}]
```

File: packages/dashmate/src/commands/update.js [hunk 1]
```diff
@@ -1,60 +1,96 @@
...
+                  }
+                } catch (e) {
+                  if (process.env.DEBUG) {
+                    console.error('Could not parse docker data', e);
+                  }
+                }
+              });
+              stream.on('error', reject);
+              stream.on('end', () => resolve({
+                serviceName, humanName, image, pulled,
```
Comment (shumkov) [inline packages/dashmate/src/commands/update.js:62]: let's call it `updated`?

File: packages/dashmate/src/docker/DockerCompose.js [hunk 2]
```diff
@@ -39,7 +41,7 @@ class DockerCompose {
   async runService(envs, serviceName, command = [], options = []) {
     await this.throwErrorIfNotInstalled();
 
-    if (await this.isServiceRunning(envs, serviceName)) {
+    if (await this.isNodeRunning(envs, serviceName)) {
```
Comment (shumkov) [inline packages/dashmate/src/docker/DockerCompose.js:44]: It's little bit wired to pass service name to `isNodeRunning` function. I would recommend to have two methods:
- `isNodeRunning` which will rely on compose files
- `isServiceRunning` which will accept service name (not optional)

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/src/listr/tasks/updateNodeTaskFactory.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+const _ = require('lodash');
```
Comment (shumkov) [inline packages/dashmate/src/listr/tasks/updateNodeTaskFactory.js:1]: It's not using Listr so it must be outside of `listr/task`. Let's create `src/update` and name it `updateServiceImagesFactory`

File: packages/dashmate/src/listr/tasks/updateNodeTaskFactory.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+                if (status?.status.includes('Image is up to date for')) {
+                  pulled = false;
+                } else if (status?.status.includes('Downloaded newer image for')) {
+                  pulled = true;
+                }
+              });
+              stream.on('error', reject);
+              stream.on('end', () => resolve({
+                serviceName, title, image, pulled,
```
Comment (shumkov) [inline packages/dashmate/src/listr/tasks/updateNodeTaskFactory.js:46]: pulled -> updated

File: packages/dashmate/src/docker/getServiceListFactory.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+function getServiceListFactory() {
+  /**
+   * Returns list of services and corresponding docker images from the config
+   * @typedef {getServiceList}
+   * @return {Config} config
+   */
+  function getServiceList(config) {
+    const services = [
+      {
+        serviceName: 'core',
```
Comment (shumkov) [inline packages/dashmate/src/docker/getServiceListFactory.js:13]: rename to `name`

File: packages/dashmate/src/commands/update.js [hunk 6]
```diff
@@ -1,60 +1,56 @@
...
 
-    try {
-      await tasks.run({
-        isVerbose,
-      });
-    } catch (e) {
-      throw new MuteOneLineError(e);
-    }
+    // Draw table or show json
+    printObject(updated
```
Comment (shumkov) [inline packages/dashmate/src/commands/update.js:30]: You should use printArrayOfObjects

File: packages/dashmate/src/printers/printObject.js [hunk 7]
```diff
@@ -9,13 +9,13 @@ const UnsupportedFormatError = require('./errors/UnsupportedFormatError');
  *
  * @param {Object} object
  * @param {string} format
+ * @param {boolean} flatten
  */
-function printObject(object, format) {
+function printObject(object, format, flatten = true) {
```
Comment (shumkov) [inline packages/dashmate/src/printers/printObject.js:14]: `printArrayOfObjects` already does it

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/src/docker/DockerCompose.js [hunk 8]
```diff
@@ -233,24 +256,37 @@ class DockerCompose {
    * Get list of Docker containers
    *
    * @param {Object} envs
-   * @param {string} [filterServiceNames]
-   * @param {boolean} returnServiceNames
-   * @return {Promise<string[]>}
+   * @param {Object} options optional
```
Comment (shumkov) [inline packages/dashmate/src/docker/DockerCompose.js:259]: Optional params must be in square brackets, also you can specify default values

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**